pygeos/geometry.py

```diff
@@ -537,3 +539,81 @@ def get_num_geometries(geometry):
     0
     """
     return lib.get_num_geometries(geometry)
+
+
+@requires_geos("3.6.0")
+@multithreading_enabled
+def get_precision(geometry):
+    """Get the precision of a geometry.
+
+    If a precision has not been previously set, it will be 0 (floating point precision).
+    Otherwise, it will return the grid size that was set on a geometry.
+
+    Returns NaN for not-a-geometry values.
+
+    Parameters
+    ----------
+    geometry : Geometry or array_like
+
+    See also
+    --------
+    set_precision
+
+    Examples
+    --------
+    >>> get_precision(Geometry("POINT (1 1)")) == 0
+    True
+    >>> geometry = set_precision(Geometry("POINT (1 1)"), 1)
+    >>> get_precision(geometry) == 1
+    True
+    >>> np.isnan(get_precision(None))
+    True
+    """
+    return lib.get_precision(geometry)
+
+
+@requires_geos("3.6.0")
+@multithreading_enabled
+def set_precision(geometry, grid_size=0, preserve_topology=False):
+    """Returns geometry with the precision set to a precision grid size.
+
+    Coordinates will be rounded if a precision grid is less precise than the input
+    geometry. Duplicated vertices will be dropped from lines and polygons for grid sizes greater
+    than 0. Line and polygon geometries may collapse to empty geometries if all vertices are
+    closer together than grid_size. Z values, if present, will not be modified.
+
+    Note: subsequent operations will always be performed in the precision
+    of the geometry with higher precision (smaller "grid_size"). That same precision will
+    be attached to the operation outputs.
+
+    Also note: input geometries should be geometrically valid; unexpected results may
+    occur if input geometries are not.
+
+    Returns None if geometry is None.
+
+    Parameters
+    ----------
+    geometry : Geometry or array_like
+    grid_size : double, optional (default: 0)
+        precision grid size.  If 0, will use floating point precision.
+    preserve_topology : bool, optional (default: False)
+        If True, will attempt to preserve the topology of a geometry after rounding
+        coordinates.
+
+    See also
+    --------
+    get_precision
+
+    Examples
+    --------
+    >>> set_precision(Geometry("POINT (0.9 0.9)"), 1)
```

Comment:
This may seem like bikeshedding, but I think it is important to signal that there is `float` expected here. It also avoids a type conversion to `float` if you give a float default.

```suggestion
    >>> set_precision(Geometry("POINT (0.9 0.9)"), 1.0)
```

Same suggestion for the default value in the function signature and docstring